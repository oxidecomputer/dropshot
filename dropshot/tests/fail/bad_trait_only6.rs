// Copyright 2024 Oxide Computer Company

#![allow(unused_imports)]

// Test for a context type with a type parameter.

#[dropshot::api_description]
trait MyApi {
    type Context<T>;
}

enum MyImpl {}

// This should not produce errors about the trait or the context type being
// missing.
impl MyApi for MyImpl {
    type Context<T> = ();
}

fn main() {
    // These items will NOT be present because of the invalid context type, and
    // will cause errors to be generated.
    my_api::api_description::<MyImpl>();
    my_api::stub_api_description();
}
