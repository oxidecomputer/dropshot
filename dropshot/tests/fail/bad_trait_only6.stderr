error: context type `Context` must not have generics
 --> tests/fail/bad_trait_only6.rs:9:17
  |
9 |     type Context<T>;
  |                 ^^^

error[E0425]: cannot find function `api_description` in module `my_api`
  --> tests/fail/bad_trait_only6.rs:23:13
   |
23 |     my_api::api_description::<MyImpl>();
   |             ^^^^^^^^^^^^^^^ not found in `my_api`

error[E0425]: cannot find function `stub_api_description` in module `my_api`
  --> tests/fail/bad_trait_only6.rs:24:13
   |
24 |     my_api::stub_api_description();
   |             ^^^^^^^^^^^^^^^^^^^^ not found in `my_api`
